### PR TITLE
Implemented backend deployment using Nginx and Gunicorn (Issue #35)

### PR DESCRIPTION
**Pull request by amulya349**

### CODE REVIEW

**Comment by amulya349** [comment]
I ran the code in same way as travis and it ran successfully. :/ 
any idea?

**Comment by inovizz** [comment]
@amulya349 - please check the Travis [build](https://travis-ci.org/Imaginea/codeXpose/builds/396273523?utm_source=github_status&utm_medium=notification) report and investigate why the build is failing. 

Also, could you please follow this guideline for writing git commit messages. 

https://chris.beams.io/posts/git-commit/

**Comment by amulya349** [comment]
Hi, as you suggested, I have changed the commit messages as per the guidelines and also, I have pushed fixes to pass the build. Please have a look and let me know if I missed anything.

File: backend/codexpose/codexpose/settings.py [hunk 1]
```diff
@@ -215,7 +215,7 @@
         'file': {
             'level': 'DEBUG',
             'class': 'logging.handlers.RotatingFileHandler',
-            'filename': BASE_DIR+'codexpose.log',
+            'filename': 'logs/codexpose.log',
```
Comment (vermaarun) [inline backend/codexpose/codexpose/settings.py]: Keep the log file in /var/log/django/codexpose/ only.

File: .travis.yml [hunk 2]
```diff
@@ -24,7 +24,8 @@ script:
   - pycodestyle codexpose/
   - pylint --rcfile=../../.pylintrc interview/
   - pycodestyle interview/
-  - touch codexpose.log
+  - mkdir -p logs
```
Comment (vermaarun) [inline .travis.yml]: Keep the log file in /var/log/django/codexpose/ only. For build in travis we have to look for any work around.

File: roles/django/tasks/main.yml [hunk 3]
```diff
@@ -8,4 +8,10 @@
 - name: Install requirements
   shell: . {{ virtualenv_dir }}/bin/activate && pip install -r {{ requirements_file }}
 
+- name: Give executable permission to start_gunicorn.sh
+  shell: chmod 777 ./start_gunicorn.sh
+
+- name: Create log directory
+  shell: mkdir -p ./backend/codexpose/logs && chown -R $(whoami) ./backend/codexpose/logs
```
Comment (vermaarun) [inline roles/django/tasks/main.yml]: Keep the log file in /var/log/ only. Change the path to /var/log/django/codexpose/

**Comment by vermaarun** [comment]
@sambasivaraop  Please take care of log file path in `settings.py`.

**Comment by vermaarun** [comment]
@amulya349 log file path in `settings.py` would be taken care in next commit. In your code also, please change the log file path to `/var/log/django/codexpose/`.

**Comment by amulya349** [comment]
@vermaarun Okay. I am making the required changes and will push a commit for that

**Comment by amulya349** [comment]
I changed the log file path to `/var/log/django/codexpose`. But, travis build is failing because to write to any file in that path, django app will need sudo permissions.

**Comment by inovizz** [comment]
@amulya349 - please take the latest pull from the master and fix your PR. The log path issue in Travis has been fixed.

**Comment by amulya349** [comment]
Closing this PR and will raise a new one with updated changes

**Pull request closed without merging**